pkg/test/framework/components/echo/util/traffic/generator.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traffic
+
+import (
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/echo"
+)
+
+const (
+	waitTimeout     = 30 * time.Second
+	defaultInterval = 1 * time.Second
+)
+
+// Config for a traffic Generator.
+type Config struct {
+	// Source of the traffic.
+	Source echo.Instance
+
+	// Options for generating traffic from the Source to the target.
+	Options echo.CallOptions
+
+	// Interval between successive call operations. If not set, defaults to 1 second.
+	Interval time.Duration
+}
+
+// Generator of traffic between echo instances.
```

Comment:
It may be helpful to explain the behavior of this. Which is, I believe, "every X seconds, send a grpc request to the client telling it to send requests to the server". Basically the idea here is to make it clear this is probably not something suitable for load tests, and is more just a convenient loop